pkg/component/controller/clusterConfig.go

```diff
@@ -102,35 +102,36 @@ func (r *ClusterConfigReconciler) Init(_ context.Context) error {
 func (r *ClusterConfigReconciler) Start(ctx context.Context) error {
 	if r.configSource.NeedToStoreInitialConfig() {
 		// We need to wait until we either succees getting the object or creating it
-		err := wait.Poll(1*time.Second, 20*time.Second, func() (done bool, err error) {
+		err := wait.Poll(1*time.Second, 20*time.Second, func() (bool, error) {
 			timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
 			defer cancel()
+
 			// Create the config object if it does not exist already
-			_, e := r.configClient.Get(timeoutCtx, constant.ClusterConfigObjectName, getOpts)
-			if e != nil {
-				if errors.IsNotFound(e) {
-					// ClusterConfig CR cannot be found, which means we can create it
-					r.log.Debugf("didn't find cluster-config object: %v", err)
-
-					if !r.leaderElector.IsLeader() {
-						r.log.Debug("I am not the leader, not writing cluster configuration")
-						return true, nil
-					}
-
-					_, e = r.copyRunningConfigToCR(ctx)
-					if e != nil {
-						r.log.Errorf("failed to save cluster-config  %v\n", err)
-						return false, nil
-					}
-				} else {
-					r.log.Errorf("error getting cluster-config: %v", err)
-					return false, nil
-				}
+			if _, err := r.configClient.Get(timeoutCtx, constant.ClusterConfigObjectName, getOpts); err == nil {
+				return true, nil
+			} else if !errors.IsNotFound(err) {
+				r.log.WithError(err).Error("Failed to get cluster configuration")
+				return false, nil
 			}
```

Comment:
I find the structure of the original code easier to read. Specifically, I don't have to mentally jump around the double-negative `!errors.IsNotFound()`, and introducing an `else if` makes me feel that there is another condition that is missing.

What about a combination of the two -- keeping the structure, but switching to the one-liner error checks and the updated logging messages?

Reply:
I think it'd be clearer if we just try to create the cluster config and check for `errors.IsAlreadyExists` instead. Saves one API call...